vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/TreeOnFailureExtension.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022 Vaadin Ltd
+ *
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.testbench.unit;
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.testbench.unit.internal.PrettyPrintTree;
+
+/**
+ * JUnit5 extension that will collect and output the component tree for the
+ * failing test UI.
+ * <p>
+ * This can help with identifying a problem that has happened
+ * in the test where a component is missing or has faulty data.
+ */
+public class TreeOnFailureExtension implements AfterTestExecutionCallback {
+
+    @Override
+    public void afterTestExecution(ExtensionContext extensionContext) {
+        boolean testFailed = extensionContext.getExecutionException()
+                .isPresent();
+        if (testFailed) {
+            final String prettyPrintTree = PrettyPrintTree.Companion.ofVaadin(
+                    UI.getCurrent()).print();
+            LoggerFactory.getLogger(TreeOnFailureExtension.class)
```

Comment:
What about using `extensionContext.publishReportEntry(...)` instead of slf4j? This way the message should be visibile on console even if logging system is not configured for test